### DISCUSSION TITLE
Any thoughts on parallel signals processing?

### DISCUSSION THREAD

**Discussion by OUCyf** [DASDAE/dascore#183] [Ideas]
I found `dascore` used `scipy` to do signal processing, some notes mentioned that we can use `jax` or others, so how is it going? Recently I used `kfr` (https://github.com/kfrlib/kfr) and `openmp` for signal processing for my ambient noise packages which are also for das processing, and from the preliminary comparison results, `kfr` is faster than scipy in detrend, filter, taper and ..., the best feelings is that it can use `openmp` in c++. 

I provide a Python interface for those functions via `xtensor`, and I am trying to publish this ambient noise package in `pypi` and `conda-forge` recently, and we can see it in a few days maybe. If it is possible I can also provide this C++ interface for `dascore` in the future.

BTW i can't use it in win64, but mac and linux is good.

**Comment by d-chambers** [comment]
Hi @OUCyf,

Sorry for the late reply, for some reason I didn't get a notification about this one.

I think it sounds great to provide a dascore interfaces for your library. Probably the easiest way is for your new package to accept a `dascore.Patch` object as a potential input, then return patch object. The `dascore.patch_function` might be useful here. 

If you don't want to take dascore as a dependency it could also be used like this:

```python
import dascore
from my_fast_dsp_library import processing_function

patch = dascore.get_example_patch()
processed_patch = patch.new(data=processing_function(patch.data))
```
 
The new routines probably can't be the default processing in DASCore, however, since we do need to support windows and scipy, although slower, is tried-and-true. 

Thanks for posting the discussion and please let me know when you publish your package, I would love to try it out.
